sips/sip-011/sip-011_ustx-marketplace-standard.md

```diff
@@ -0,0 +1,427 @@
+
+### Marketplace-Trait function 2: Unlist in ustx
+`(unlist-in-ustx (uint) (response bool uint))`
+
+This function announces the unlisting to a global marketplace. This must return `(ok true)` on success, never `(ok false)` and it must send a unlist event.
+
+The Function takes a token id or token amount to be unlisted.
+
+### Marketplace-Trait function 3: Buy in ustx and annonce unlisting
+`(buy-in-ustx (uint <commission-trait>) (response bool uint))`
+
+This function buys the listed token for the selling price and unlists it.
+
+This function takes an amount of the token (selling price). Additionally the function takes a commission-trait as defined below it will dictate actions to happen after the sale (owner, artist, marketplace, etc. getting their share of the sale).
+
+This function must be defined with define-public, as it alters state, and must be externally callable. The function must return `(ok true)` on success, never `(ok false)`.
+
+### Marketplace-Trait function 4: Get asset
+`(get-asset () (response {fq-contract: string, asset-class: string} uint))`
+
+The is a read only function defining the asset.
+
+
+### Commision-Trait function:
+`(pay (uint uint) (response bool uint))`
+
+An additional action after the marketplace sale happened. Usually a token transfer to the marketplace.
+
+The function takes an identifier of an NFT and and the price in micro STX (uSTX). The function must return `(ok true)` on success, never `(ok false)`
+
+Optionally seperate out royalties, as in the reference: See from line 440.
+
+It is recommended to use error codes from standardized list of codes and implement the function for converting the error codes to messages function that are defined in a separate SIP.
+
+## Marketplace-Trait-and-Commision-Trait
+
+```
+(use-trait commission-trait .commisions.trait)
+(define-trait marketplace
+    (
+        (list-in-ustx (uint uint <commission-trait>) (response bool uint))
+        (unlist-in-ustx (uint) (response bool uint))
+        (buy-in-ustx (uint  <commission-trait>) (response bool uint))
+        (get-asset () (response {fq-contract: string, asset-class: string} uint))
+    )
+)
+
+(define-trait commission
+    (
+        (pay (uint uint) (response bool uint))
+    )
+)
+```
+
+## Security
+As `commission-traits` can call any functions in the name of the tx-sender, it is important that a web app only offers commission contracts that are well understood. In particular, appropriate post-conditions have to be created.
+
+If asset contracts want to control trades they have to restrict which operators are approved. Note, that royalties to an artist of an NFT can be part of the commission if agreed with the marketplace. They can also be implemented in the NFT directly or as a seperate trait. See reference implementation list-in-ustx.
+
+## Use of native asset functions
+Werner: not sure what to do here (section just a copy from sip009)
```

Comment:
IMO you could make a reference to SIP-009 (with a link) along with a short sentence to mention they should follow the same recommendations to avoid repeating content.